extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-config.js

```diff
@@ -231,5 +223,92 @@ describes.realWin(
         ).to.have.been.calledOnce;
       });
     });
+
+    it('test validate required fields for config', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+
+      for (const [key, value] of Object.entries(productValidationConfig)) {
+        if (value.includes(validateRequired)) {
+          delete invalidConfig['items'][0][key];
+        }
+      }
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      for (const [key, value] of Object.entries(productValidationConfig)) {
+        if (value.includes(validateRequired)) {
+          const errorString = `Error: Field ${key} is required.`;
+          expect(spy).to.have.been.calledWith(
+            'AMP-STORY-SHOPPING-CONFIG',
+            errorString
+          );
+        }
+      }
+    });
+
+    it('should fail config validation because an expected string JSON value is of a non-string type', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      invalidConfig['items'][0]['productTitle'] = 50; // This value is not a string
+
+      const errorString = `Error: productTitle ${invalidConfig['items'][0]['productTitle']} is not a string`;
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected string JSON value is not a valid HTML id', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      invalidConfig['items'][0]['productId'] = '1234city-pop'; // productId starts iwth a number, so it is an invalid HTML Id
+
+      const errorString = `Error: productId ${invalidConfig['items'][0]['productId']} is not a valid HTML Id`;
```

Comment:
In this errorString and the errorStrings in the tests below, feel free to write out the key name instead of using these template placeholders because the latter takes up more space and is less readable

"Error: productId 1234city-pop is not a valid HTML Id"
instead of 
"Error: productId ${invalidConfig['items'][0]['productId']} is not a valid HTML Id"